rskj-core/src/main/java/org/ethereum/core/SignatureCache.java

```diff
@@ -20,27 +20,35 @@
 package org.ethereum.core;
 
 import co.rsk.core.RskAddress;
+import co.rsk.crypto.Keccak256;
 import co.rsk.remasc.RemascTransaction;
+import co.rsk.util.MaxSizeHashMap;
 
+import java.util.Collections;
 import java.util.Map;
 
 public abstract class SignatureCache {
-    protected Map<Transaction, RskAddress> addressesCache;
 
-    public abstract RskAddress getSender(Transaction transaction);
+    protected Map<Keccak256, RskAddress> addressesCache;
 
-    public boolean containsTx(Transaction transaction) {
-        return addressesCache.containsKey(transaction);
+    public SignatureCache(int maxCacheSize, boolean accessOrder) {
+        addressesCache = Collections.synchronizedMap(
```

Comment:
just thinking that we may not need this collection to be synchronized anymore?

Reply:
If we're. enforcing the sync via the method modifiers, we can get rid of this I think, yes.

Reply:
Updated!